src/components/product-filters/product-filters.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+import { useNavigation } from '@remix-run/react';
+import { IProductFilters } from '~/api/types';
+import { Accordion } from '../accordion/accordion';
+import { RangeSlider } from '../range-slider/range-slider';
+
+interface ProductFiltersProps {
+    appliedFilters: IProductFilters;
+    onFiltersChange: (filters: IProductFilters) => void;
+    lowestPrice: number;
+    highestPrice: number;
+    formatPrice?: (value: number) => string;
+}
+
+export const ProductFilters = ({
+    appliedFilters,
+    onFiltersChange,
+    lowestPrice,
+    highestPrice,
+    formatPrice,
+}: ProductFiltersProps) => {
+    const navigation = useNavigation();
+
+    const [filters, setFilters] = useState(appliedFilters);
+
+    const handleFiltersChange = (changed: Partial<IProductFilters>) => {
+        const newFilters = { ...filters, ...changed };
+        setFilters(newFilters);
+        onFiltersChange(newFilters);
+    };
+
+    // Synchronize filters on back/forward browser button clicks.
+    useEffect(() => {
+        if (navigation.state !== 'loading') {
+            setFilters(appliedFilters);
+        }
+    }, [navigation.state, appliedFilters]);
+
+    return (
+        <Accordion
+            small
+            items={[
```

Comment:
Unrelated to the PR, but I find providing items as children more intuitive / maybe easier to work with in Codux?

https://www.radix-ui.com/primitives/docs/components/accordion

Unfortunately their accordion doesn't change overflow to visible when expanded. (The MUI accordion does.)

Reply:
I'll take a look if I can do it with Radix, because I want to use Radix select for product variants.